Upgrade Agones to 1.29.0 and Open Match to 1.7.0

govargo opened the pull request:
This PR has a upgrade to Agones and Open Match and change of installing for Open Match

- Upgrade GKE Kubernetes version from 1.22 to 1.24
- Upgrade Agones from 1.27.0 to 1.29.0
- Upgrade Open Match from 1.5.0 to 1.7.0
- Change instaling way for Open Match ※1

※1 One of the Open Match installing way is YAML apply. The default YAML requires more than 4.5 cpu cores, because the StatefulSet's Redis has 3 containers(redis, sentinel, metrics) requre each 0.5 cpus(= total 1.5 cores) and StatefulSet has 3 reqlica(= 1.5cores * 3Pod = total 4.5 cores)

govargo commented:
I'll assign review after 3/7.

kemurayama commented on Makefile:
```diff
@@ -165,8 +171,8 @@ openmatch-uninstall-local:
 openmatch-uninstall:
-	kubectl delete psp,clusterrole,clusterrolebinding --selector=release=$(OM_NS)
-	kubectl delete namespace $(OM_NS)
+	helm uninstall -n ${OM_NS} ${OM_NS}
```
Hi @govargo, 

Would you add `helm repo remove` instruction or new `make` commands for `helm repo add` and `helm repo remove` ?

govargo replied:
Added `helm repo remove` to Makefile.

kemurayama replied:
LGTM

kemurayama reviewed (changes requested):
Hi @govargo 

As helm is introduced, it is now the necessary tool for Space Agon. 

Would you revise README.md in accordance with your PR ?

https://github.com/googleforgames/space-agon/blob/e333e5eaa75efeb5bb76b863d69b91298d47c7ab/README.md?plain=1#L28

govargo commented:
> Would you revise README.md in accordance with your PR ?

I added the helm section.
I think helm is the neccesary, not optional. So, I removed (Optional).

govargo commented:
@kemurayama 
What do you mean about `need to remove the duplicate line.`?
Could you please point out the line with link?

kemurayama commented:
Hi @govargo I mentioned this line. Thanks.

https://github.com/govargo/space-agon/blob/b49c440a0f32ac405bd40609d19c695bc4bb1615/README.md?plain=1#L27

govargo commented:
Thank you for your point out @kemurayama.

It's my fault. I fixed this duplicate line.

kemurayama reviewed (approved):
Hi @govargo, thanks for updating. LGTM
